plugins/core/reporter/pprof_manager.go

```diff
@@ -0,0 +1,330 @@
+	StartTask() (io.Writer, error)
+	StopTask(io.Writer)
+	IsDirectSamplingType() bool
+}
+type PprofReporter interface {
+	ReportPprof(taskID string, content []byte)
+}
+
+var NewPprofTaskCommand func(taskID, events string, duration time.Duration,
+	createTime int64, dumpPeriod int, pprofFilePath string,
+	logger operator.LogOperator, manager PprofReporter) PprofTaskCommand
+
+type PprofTaskManager struct {
+	logger         operator.LogOperator
+	serverAddr     string
+	pprofInterval  time.Duration
+	PprofClient    pprofv10.PprofTaskClient // for grpc
+	connManager    *ConnectionManager
+	entity         *Entity
+	pprofFilePath  string
+	LastUpdateTime int64
+	commands       PprofTaskCommand
+	pprofSendCh    chan *pprofv10.PprofData
+}
+
+func NewPprofTaskManager(logger operator.LogOperator, serverAddr string,
+	pprofInterval time.Duration, connManager *ConnectionManager,
+	pprofFilePath string) (*PprofTaskManager, error) {
+	pprofManager := &PprofTaskManager{
+		logger:        logger,
+		serverAddr:    serverAddr,
+		pprofInterval: pprofInterval,
+		connManager:   connManager,
+		pprofFilePath: pprofFilePath,
+		pprofSendCh:   make(chan *pprofv10.PprofData, maxPprofSendQueueSize),
+	}
+	if pprofInterval > 0 {
+		conn, err := connManager.GetConnection(serverAddr)
+		if err != nil {
+			return nil, err
+		}
+		pprofManager.PprofClient = pprofv10.NewPprofTaskClient(conn)
+		pprofManager.commands = nil
+	}
+	return pprofManager, nil
+}
+
+func (r *PprofTaskManager) InitPprofTask(entity *Entity) {
+	if r.PprofClient == nil {
+		return
+	}
+	r.entity = entity
+	r.initPprofSendPipeline()
+	go func() {
+		for {
+			switch r.connManager.GetConnectionStatus(r.serverAddr) {
+			case ConnectionStatusShutdown:
+				return
+			case ConnectionStatusDisconnect:
+				time.Sleep(r.pprofInterval)
+				continue
+			}
+			pprofCommand, err := r.PprofClient.GetPprofTaskCommands(context.Background(), &pprofv10.PprofTaskCommandQuery{
+				Service:         r.entity.ServiceName,
+				ServiceInstance: r.entity.ServiceInstanceName,
+				LastCommandTime: r.LastUpdateTime,
+			})
+			if err != nil {
+				r.logger.Errorf("fetch pprof task commands error %v", err)
+				time.Sleep(r.pprofInterval)
+				continue
+			}
+
+			if len(pprofCommand.GetCommands()) > 0 && pprofCommand.GetCommands()[0].Command == "PprofTaskQuery" {
+				rawCommand := pprofCommand.GetCommands()[0]
+				r.HandleCommand(rawCommand)
+			}
+
+			time.Sleep(r.pprofInterval)
+		}
+	}()
+}
+
+func (r *PprofTaskManager) HandleCommand(rawCommand *commonv3.Command) {
+	command := r.deserializePprofTaskCommand(rawCommand)
+	if command.GetCreateTime() > r.LastUpdateTime {
+		r.LastUpdateTime = command.GetCreateTime()
+	} else {
+		return
+	}
+
+	if command.IsDirectSamplingType() {
+		// direct sampling of Heap, Allocs, Goroutine, Thread
+		writer, err := command.StartTask()
+		if err != nil {
+			r.logger.Errorf("start %s pprof task error %v \n", command.GetTaskID(), err)
+			return
+		}
+		command.StopTask(writer)
```

Comment:
Could the agent add some log when the task starts/stops, it could help quickly know the task execution status.